diff --git a/package.json b/package.json
index 0ca6e73a4ff68b4a36431ff81c92f3a2556b622b..7969dd1d42881e7e91a20b6248cdb7a6ae9455e8 100644
--- a/package.json
+++ b/package.json
@@ -26,6 +26,7 @@
   "license": "MIT",
   "homepage": "https://shopify.github.io/flash-list/",
   "main": "dist/index.js",
+  "react-native": "src/index.ts",
   "types": "dist/index.d.ts",
   "scripts": {
     "up": "bundle install && yarn fixture:rn:up && yarn e2e:up && yarn build",
diff --git a/src/FlashList.tsx b/src/FlashList.tsx
index 0e9f071a53d113e85f3099ccf239d1f3d06361b6..0d8da0f444a5cc291621732369e71e0b693b4ff3 100644
--- a/src/FlashList.tsx
+++ b/src/FlashList.tsx
@@ -4,6 +4,7 @@ import {
   RefreshControl,
   LayoutChangeEvent,
   NativeSyntheticEvent,
+  ScrollViewComponent,
   StyleSheet,
   NativeScrollEvent,
 } from "react-native";
@@ -842,6 +843,12 @@ class FlashList<T> extends React.PureComponent<
     return this.rlvRef?.getScrollableNode?.() || null;
   }
 
+  public getNativeScrollRef(): React.ElementRef<
+    typeof ScrollViewComponent | null
+  > {
+    return this.rlvRef?.getNativeScrollRef?.() || null;
+  }
+
   /**
    * Allows access to internal recyclerlistview. This is useful for enabling access to its public APIs.
    * Warning: We may swap recyclerlistview for something else in the future. Use with caution.
