diff --git a/lib/module/layoutReanimation/web/componentUtils.js b/lib/module/layoutReanimation/web/componentUtils.js
index afd8f89be6e0a073b461c9c00704dbfd8852c5e5..10645c2822e960ff9911775cc267f294c2a6d1ed 100644
--- a/lib/module/layoutReanimation/web/componentUtils.js
+++ b/lib/module/layoutReanimation/web/componentUtils.js
@@ -114,6 +114,7 @@ export function setElementAnimation(element, animationConfig, shouldSavePosition
     element.style.animationDuration = `${duration}s`;
     element.style.animationDelay = `${delay}s`;
     element.style.animationTimingFunction = easing;
+    element.style.animationFillMode = "backwards";
   };
   if (animationConfig.animationType === LayoutAnimationType.ENTERING) {
     // On chrome sometimes entering animations flicker. This is most likely caused by animation being interrupted
diff --git a/src/layoutReanimation/web/componentUtils.ts b/src/layoutReanimation/web/componentUtils.ts
index 6131c43bb1bf8724d54112c9e316ade0e2840d97..de789772e5f16df9eef40112c54f098d1e66c073 100644
--- a/src/layoutReanimation/web/componentUtils.ts
+++ b/src/layoutReanimation/web/componentUtils.ts
@@ -179,6 +179,7 @@ export function setElementAnimation(
     element.style.animationDuration = `${duration}s`;
     element.style.animationDelay = `${delay}s`;
     element.style.animationTimingFunction = easing;
+    element.style.animationFillMode = "backwards";
   };
 
   if (animationConfig.animationType === LayoutAnimationType.ENTERING) {
