diff --git a/lib/module/layoutReanimation/web/componentUtils.js b/lib/module/layoutReanimation/web/componentUtils.js
index c3623f90862b4413ce6d77bb04496ad2fecb3125..fc64b7906cefb041de25e859f3071f4413054d59 100644
--- a/lib/module/layoutReanimation/web/componentUtils.js
+++ b/lib/module/layoutReanimation/web/componentUtils.js
@@ -109,6 +109,7 @@ export function setElementAnimation(element, animationConfig, shouldSavePosition
     element.style.animationDuration = `${duration}s`;
     element.style.animationDelay = `${delay}s`;
     element.style.animationTimingFunction = easing;
+    element.style.animationFillMode = 'backwards';
   };
   if (animationConfig.animationType === LayoutAnimationType.ENTERING) {
     // On chrome sometimes entering animations flicker. This is most likely caused by animation being interrupted
diff --git a/src/layoutReanimation/web/componentUtils.ts b/src/layoutReanimation/web/componentUtils.ts
index 53bfae44e3dfee63f2fbc86566f99d1f586396d1..55de122b3c4dbb4208c5c3101d63a1444fe31734 100644
--- a/src/layoutReanimation/web/componentUtils.ts
+++ b/src/layoutReanimation/web/componentUtils.ts
@@ -163,6 +163,7 @@ export function setElementAnimation(
     element.style.animationDuration = `${duration}s`;
     element.style.animationDelay = `${delay}s`;
     element.style.animationTimingFunction = easing;
+    element.style.animationFillMode = 'backwards';
   };
 
   if (animationConfig.animationType === LayoutAnimationType.ENTERING) {
